### PR TITLE
EZP-31519: Move app.php rejection snippet from front controller to ezpublish-kernel

### PR DESCRIPTION
**Pull request by webhdx**
| Question           | Answer
| ------------------ | ------------------
| **JIRA issue**     | [EZP-31519](https://jira.ez.no/browse/EZP-31519)
| **Improvement**| yes
| **New feature**    | yes/no (it's new to the kernel but not to the application)
| **Target version** | `6.13`
| **BC breaks**      | no
| **Tests pass**     | yes
| **Doc needed**     | no

**TODO**:
- [x] Implement feature / fix a bug.
- [x] Implement tests.
- [x] Fix new code according to Coding Standards (`$ composer fix-cs`).
- [x] Ask for Code Review.

### CODE REVIEW

File: eZ/Bundle/EzPublishCoreBundle/EventListener/RejectExplicitFrontControllerRequestsListener.php [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+     * @param \Symfony\Component\HttpKernel\Event\GetResponseEvent $event
+     */
+    public function onKernelRequest(GetResponseEvent $event)
+    {
+        if ($event->getRequestType() !== HttpKernelInterface::MASTER_REQUEST) {
+            return;
+        }
+
+        $request = $event->getRequest();
+        $scriptFileName = preg_quote(basename($request->getScriptName()), '\\');
```
Comment (glye) [inline eZ/Bundle/EzPublishCoreBundle/EventListener/RejectExplicitFrontControllerRequestsListener.php]: Could you expand on "better compatibility" please, for educational purposes?
I was going to say that basename() isn't needed anymore, but it seems that SCRIPT_NAME includes the leading slash, so let's keep it.
(Both SCRIPT_NAME and SCRIPT_FILENAME are injection safe, from what I've read.)

Comment (webhdx) [reply eZ/Bundle/EzPublishCoreBundle/EventListener/RejectExplicitFrontControllerRequestsListener.php]: If you look at the implementation of `Request::getScriptName` you will see that it's resolved from `SCRIPT_NAME` or `ORIG_SCRIPT_NAME` for compatibility with 1and1 hosting provider. I think it's safer to use it this way.

**Comment by webhdx** [comment]
@mnocon Ready for retesting

**Comment by lserwatka** [comment]
@webhdx could you merge it up to 7.5 and ezplatform-kernel/master

**Pull request merged**